discord_bot/discord_bot.py

```diff
@@ -101,29 +127,33 @@ def get_gateway(self, **kwargs):
                                                                     self.config["gateway_encoding"])) as websocket:
             self.websocket = websocket
             while True:
-                response = await self.websocket.recv()
-                response = json.loads(response)
-                print("{}: {}".format(Opcodes(response["op"]).name, response))
-                if response["op"] == Opcodes.HELLO:
-                    asyncio.ensure_future(self.handshake(response))
-                elif response["op"] == Opcodes.HEARTBEAT_ACK:
+                message = await self.websocket.recv()
+                message = json.loads(message)
+                print("{}: {}".format(Opcodes(message["op"]).name, message))
+                if message["op"] == Opcodes.HELLO:
+                    asyncio.ensure_future(self.handshake(message))
+                elif message["op"] == Opcodes.HEARTBEAT_ACK:
                     pass
-                elif response["op"] == Opcodes.INVALID_SESSION:
+                elif message["op"] == Opcodes.INVALID_SESSION:
+                    self.last_seq = message["s"]
```

Comment:
Just move `last_seq` parsing out of all of the opcode control flow; it's too messy to mix two things that have nothing to do with each other.